litestar/response/template.py

```diff
@@ -99,7 +100,16 @@ def to_asgi_response(
         status_code: int | None = None,
         type_encoders: TypeEncodersMap | None = None,
     ) -> ASGIResponse:
-        if not app.template_engine:
+        if app is not None:
+            warn_deprecation(
+                version="2.1",
+                deprecated_name="app",
+                kind="parameter",
+                removal_in="3.0.0",
+                info="app will be pulled from parameter 'request' instead",
+            )
+
+        if not request.app.template_engine:
```

Comment:
Maybe assign `app` to a separate variable here instead of accessing it via `request.app` each time.